model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -358,7 +358,7 @@ def __init__(
         interpolation_state: diffusion_states.DiffusionInterpolationState,
         edge_params: grid_states.EdgeParams,
         cell_params: grid_states.CellParams,
-        backend: gtx_backend.Backend | None,
+        backend: gtx.backend.Backend | Literal["gpu", "cpu"] | dict[str, Any] | None = None,
```

Comment:
- we should use an enum for gpu/cpu or something similar -> see slack discussion
- I would propose to introduce an type alias for `dict[str, Any]` to help the reader, e.g. `BackendDescription: TypeAlias = dict[str, Any]`

Reply:
how would that help the reader?